prow/cmd/exporter/collector.go

```diff
@@ -169,3 +171,153 @@ func sanitizeLabelName(s string) string {
 func escapeString(v string) string {
 	return escapeWithDoubleQuote.Replace(v)
 }
+
+type prowJobLifecycleCollector struct {
+	lastKnownState map[types.UID]lastState
```

Comment:
I think we need some kind of garbage collection for this, else the exporter can not run too long before getting oom killed. Maybe just something that removes all jobs from this that don't exist anymore?

Reply:
This is garbage-collected. See the `Delete` callback. It worst it still requires less space than the informer cache of the jobs.

I am more concerned regarding to the historams themselves. Over time we will have more and more histograms, which probably means that we push obsolete histograms and need more and more memory. I am not yet entirely sure on how to best handle it.